### DISCUSSION TITLE
How to get request post data?

### DISCUSSION THREAD

**Discussion by xiangnanscu** [nuxt/nuxt#16811] [Questions] [answered]
file  `server/api/env.ts` , and I `curl -X POST -H "Content-Type: application/json"  -d '{"name": "xn", "email": "xn@example.com"}'  http://localhost:3000/api/env`

```
//server/api/env.ts
export default defineEventHandler((event) => {
  console.log(event); // how to get post data here ?
  return event
})
```

**Comment by Drider** [comment, marked as answer]
Hello!
I think you should use something like this:
```
import { readBody } from 'h3';

export default defineEventHandler(async (event) => {
  const body = await readBody(event);

  console.log(body);

  return body;
});
```
[h3 readBody](https://www.jsdocs.io/package/h3#readBody)

See also examples in the [documentation](https://v3.nuxtjs.org/guide/directory-structure/server)
